content/blog/scaleway-psp.md

```diff
@@ -0,0 +1,176 @@
+après plus de 10 minutes... Étrange.
+
+```console
+NAME                                             STATUS     ROLES    AGE   VERSION
+scw-particule-commonpool-dbfe057a9ee64e56b50f9   NotReady   <none>   11m   v1.18.2
+```
+
+Le problème est qu'activer les PodSecurityPolicy ça ne suffit pas, pire ça
+"casse" le cluster si rien d'autre n'est effectué.
+
+Comme nous l'avons déjà expliqué, une fois les PSP activées, tous les pods qui
+souhaiterons être schédulés devront fournir une PSP à l'API Server afin que
+celui ci valide qu'ils respectent bien les règles. Si aucune PSP n'est
+disponible ou qu'un pod n'a le droit d'accéder à aucunes d'entre elles, l'API
+Server lui refusera le déploiement.
+
+Et c'est exactement ce qu'il se passe chez Kapsule. Personne ne peut se
+schéduler car :
+
+- Aucune PSP n'est présente
+- Aucun droit n'est donné pour utiliser une PSP
+
+On va donc corriger cela en créant une PSP `privileged` qui permettra
+d'absolument tout faire (comme si il n'y avait pas de PSP en fin de compte) et
+autoriser l'ensemble du cluster à utiliser cette PSP.
+
+```yaml
+---
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: privileged
+spec:
+  allowPrivilegeEscalation: true
+  allowedCapabilities:
+  - '*'
+  allowedUnsafeSysctls:
+  - '*'
+  fsGroup:
+    rule: RunAsAny
+  hostIPC: true
+  hostNetwork: true
+  hostPID: true
+  hostPorts:
+  - max: 65535
+    min: 0
+  privileged: true
+  runAsUser:
+    rule: RunAsAny
+  seLinux:
+    rule: RunAsAny
+  supplementalGroups:
+    rule: RunAsAny
+  volumes:
+  - '*'
+---
+```
+
+Puis le ClusterRole et le ClusterRoleBinding qui permettront à tous les
+éléments authentifiés de votre cluster d'utiliser (`use`) la PSP :
+
+```yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: psp:privileged
+rules:
+- apiGroups:
+  - policy
+  resourceNames:
+  - privileged
+  resources:
+  - podsecuritypolicies
+  verbs:
+  - use
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: psp:any:privileged
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: psp:privileged
+subjects:
+- apiGroup: rbac.authorization.k8s.io
+  kind: Group
+  name: system:authenticated
+```
+
+On applique tout ce petit monde et on attend tranquillement que les pods CNI et
+kube-proxy se déploient pour que notre node passe en `Ready`.
+
+```console
+NAME                                             STATUS   ROLES    AGE    VERSION
+scw-particule-commonpool-dbfe057a9ee64e56b50f9   Ready    <none>   17m   v1.18.2
+```
+
+### Conclusion
+
+Il est évidemment important de noter qu'il ne s'agit pas d'une faute de
+configuration de Scaleway. Le composant demandé est bien appliqué, il manque
+seulement certaines ressources que n'importe quel utilisateur peut créer pour
+profiter pleinement de la fonctionnalité. On pourra aussi remarquer que GKE ne
```

Comment:
Not sur about GKE but EKS does not, maybe link to my issue here https://github.com/aws/containers-roadmap/issues/401